cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -187,32 +187,40 @@ private Boolean isPrewarmed(Long clusterId) {
     public void restartStaleServices(MgmtServiceResourceApi mgmtServiceResourceApi, ClustersResourceApi clustersResourceApi)
             throws ApiException, CloudbreakException {
         restartClouderaManagementServices(mgmtServiceResourceApi);
-        restartCMStaleServices(clustersResourceApi);
+        deployConfigAndRefreshCMStaleServices(clustersResourceApi);
     }
 
-    private void restartCMStaleServices(ClustersResourceApi clustersResourceApi) throws ApiException, CloudbreakException {
+    private void deployConfigAndRefreshCMStaleServices(ClustersResourceApi clustersResourceApi) throws ApiException, CloudbreakException {
         LOGGER.debug("Restarting stale services and redeploying client configurations in Cloudera Manager.");
         ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(apiClient);
         List<ApiService> services = servicesResourceApi.readServices(stack.getName(), SUMMARY).getItems();
         boolean configStale = services.stream().anyMatch(service -> !ApiConfigStalenessStatus.FRESH.equals(service.getConfigStalenessStatus()));
-        if (configStale) {
-            Optional<ApiCommand> optionalRestartCommand = clustersResourceApi.listActiveCommands(stack.getName(), SUMMARY).getItems().stream()
-                    .filter(cmd -> "RestartWaitingForStalenessSuccess".equals(cmd.getName())).findFirst();
-            ApiCommand restartServicesCommand;
-            if (optionalRestartCommand.isPresent()) {
-                restartServicesCommand = optionalRestartCommand.get();
-                LOGGER.debug("Restart for stale services is already running with id: [{}]", restartServicesCommand.getId());
-            } else {
-                restartServicesCommand = clustersResourceApi.restartCommand(stack.getName(),
-                        new ApiRestartClusterArgs().redeployClientConfiguration(Boolean.TRUE).restartOnlyStaleServices(Boolean.TRUE));
-            }
-            pollRestart(restartServicesCommand);
-            LOGGER.debug("Restarted stale services in Cloudera Manager.");
+        boolean clientConfigStale = services.stream().anyMatch(service -> !ApiConfigStalenessStatus.FRESH.equals(service.getClientConfigStalenessStatus()));
+        if (configStale || clientConfigStale) {
```

Comment:
what's behind the `configStale` and the `clientConfigStale`? the actual values come from a specific logic but the variable name does not reflect it. let's say something like `serviceContainsStatusWithNotFreshStaleness`.. or other.. it's much longer but a good variable name could make unnecessary to understand the stream-filter line(s) above.

also, do we have a unit-test for this logic somewhere? if not please make some.

Reply:
I think this variable name too long. The other hand it just a boolean flag, not a list of services with not fresh staleness.

Reply:
surely it's too long, I didn't say that you have to use my suggestion, but if you read this condition for the first time you must read the assignation line to understand what the variable actually represents. that's why it's a bad choice of name.